packages/react-components/react-utilities/src/hooks/useOnClickOutside.ts

```diff
@@ -96,26 +98,26 @@ export const useOnClickOutside = (options: UseOnClickOrScrollOutsideOptions) =>
     };
 
     // use capture phase because React can update DOM before the event bubbles to the document
-    element?.addEventListener('click', conditionalHandler, true);
-    element?.addEventListener('touchstart', conditionalHandler, true);
-    element?.addEventListener('contextmenu', conditionalHandler, true);
-    element?.addEventListener('mousedown', handleMouseDown, true);
+    element.addEventListener('click', conditionalHandler, true);
+    element.addEventListener('touchstart', conditionalHandler, true);
+    element.addEventListener('contextmenu', conditionalHandler, true);
+    element.addEventListener('mousedown', handleMouseDown, true);
 
     // Garbage collect this event after it's no longer useful to avoid memory leaks
-    timeoutId.current = win?.setTimeout(() => {
+    const timer = win?.setTimeout(() => {
       currentEvent = undefined;
     }, 1);
 
     return () => {
-      element?.removeEventListener('click', conditionalHandler, true);
-      element?.removeEventListener('touchstart', conditionalHandler, true);
-      element?.removeEventListener('contextmenu', conditionalHandler, true);
-      element?.removeEventListener('mousedown', handleMouseDown, true);
+      element.removeEventListener('click', conditionalHandler, true);
+      element.removeEventListener('touchstart', conditionalHandler, true);
+      element.removeEventListener('contextmenu', conditionalHandler, true);
+      element.removeEventListener('mousedown', handleMouseDown, true);
 
-      win?.clearTimeout(timeoutId.current);
+      win?.clearTimeout(timer);
       currentEvent = undefined;
     };
-  }, [listener, element, disabled, handleMouseDown, win]);
+  }, [element, disabled, win]);
```

Comment:
This is not so shiny:

<img width="1107" alt="image" src="https://github.com/user-attachments/assets/cff5c870-35c2-462a-93ea-5c84c4eb3b76" />

AFAIR that was the reason (unstable deps) to use `useEventCallback` for listener

Reply:
For context:

<img width="725" alt="image" src="https://github.com/user-attachments/assets/cb53dd7b-1d1b-40d6-858e-bea4b75ec5e8" />

Reply:
> to avoid unnecessary hooks

Well, they are necessary to avoid updating subscriptions on every render as `useEventCallback` returns a stable callback.

- With your refactor (in the current state) we will get stale values in the subscription as `callback` & etc. are not in dependencies
- If `callback` & etc. will be in dependencies ➡️ `useEffect` will be called on every render 💥

Reply:
However, when the rerender actually happens, the previous `useEffect` will just do the disposal and remove all the event listeners, then in the new `useEffect`, the event listeners are added back again.

In this case, it really doesn't matter whether the previous and current functions are the same reference.
What we should avoid is the unnecessary `dispose and recreate`.

For my changes, the dependencies are less than before, tending to be more stable.

Reply:
I'll investigate more in a new PR. thanks.